app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt

```diff
@@ -1,52 +1,135 @@
+        var secondValue = (secondOperandUtils.getValue()).toString()
+        if (secondOperandUtils.isEmpty()) {
+            secondValue = EMPTY_STRING
+        }
+        return "$firstValue$operator$secondValue"
+    }
 
-    override fun getFullResult(): String {
-        var firstOperandValue = firstOperand.toDouble()
-        var secondOperandValue = secondOperand.toDouble()
+    override fun getPartialResult(): String {
+        return getValue()
+    }
 
-        return when (operator) {
-            OPERATOR_SUM -> (firstOperandValue + secondOperandValue).toString()
-            OPERATOR_MINUS -> (firstOperandValue - secondOperandValue).toString()
-            OPERATOR_DIVIDE -> (firstOperandValue / secondOperandValue).toString()
-            OPERATOR_MULTIPLY -> (firstOperandValue * secondOperandValue).toString()
-            else -> ERROR_MESSAGE
+    private fun isValidOperation(): Boolean {
+        if (operator.isEmpty()) {
+            if (firstOperandUtils.isEmpty()) {
+                result = EMPTY_STRING
+            } else {
+                result = java.lang.String.valueOf(firstOperandUtils.getValue())
+            }
+            return false
+        } else if (secondOperandUtils.isEmpty()) {
+            error = ErrorUtils.ERROR_MESSAGE_INVALID_FORMAT
+            result = EMPTY_STRING
+            return false
         }
+        return true
+    }
+
+    override fun doOperations() {
+        if (isValidOperation()) {
+
+            when (operator) {
+                OPERATOR_SUM -> {
+                    result =
+                        (firstOperandUtils.getValue() + secondOperandUtils.getValue()).toString()
+                    error = ErrorUtils.NONE
+                }
+                OPERATOR_MINUS -> {
+                    result =
+                        (firstOperandUtils.getValue() - secondOperandUtils.getValue()).toString()
+                    error = ErrorUtils.NONE
+                }
+                OPERATOR_DIVIDE -> if (secondOperandUtils.getValue() == ZERO_NUMBER_DOUBLE_TYPE) {
+                    result = EMPTY_STRING
+                    error = ErrorUtils.ERROR_MESSAGE_DIVISION
+                } else {
+                    result =
+                        (firstOperandUtils.getValue() / secondOperandUtils.getValue()).toString()
+                    error = ErrorUtils.NONE
+                }
+                OPERATOR_MULTIPLY -> {
+                    result =
+                        (firstOperandUtils.getValue() * secondOperandUtils.getValue()).toString()
+                    error = ErrorUtils.NONE
+                }
+                else -> {
+                    ErrorUtils.ERROR_MESSAGE.toString()
+                    result = EMPTY_STRING
+                }
+
+            }
+            updateFirstOperand()
+            secondOperandUtils.eraseOperands()
+
+        }
+    }
+
+    override fun getResult(): String {
+        return result
+    }
```

Comment:
if you only have to return a value you can use
    override fun getResult(): String = result

Reply:
Thx for the tip, now its done